src/quick-lint-js/error.h

```diff
@@ -1139,6 +1139,12 @@
       { source_code_span continue_statement; },                                \
       .error(QLJS_TRANSLATABLE("continue can only be used inside of a loop"),  \
              continue_statement))                                              \
+                                                                               \
+  QLJS_ERROR_TYPE(                                                             \
+      error_redundant_semicolon_after_else, "E202",                            \
+      { source_code_span semicolon; },                                         \
+      .warning(QLJS_TRANSLATABLE("redundant semicolon after else"),            \
```

Comment:
Blocking: This error message is incorrect. The extra semicolon is *not* redundant; it changes behavior.

Example:
```js
for (let b of [true, false]) {
  if (b) {
    console.log('t');
  } else /* Maybe add a semicolon here */ {
    console.log('f');
  }
  console.log('---');
}
```
Without semicolon:
```
t
---
f
---
```
With semicolon:
```
t
f
---
f
---
```

We should communicate that the semicolon is likely unintended, but also imply that removing the semicolon would change behavior.

Perhaps this message is better: "else has empty body; consider removing the semicolon or removing the 'else'" The more I think about it, the more I think we should have two different error messages:

```js
if (b) {
} else; {  // else has empty body; consider removing the semicolon to make it not empty
}
```
```js
if (b) {
} else;   // else has empty body; consider removing the redundant 'else'
{
}
```
(The first example has no line break between `;` and `{`. The second example does.)

(Perhaps the second case doesn't deserve a diagnostic at all though. Or perhaps it should be of a lesser severity than "warning".)